tests/block2tests.c

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ *
+ * Copyright (c) 2023 GARDENA GmbH
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * The Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ *   Lukas Woodtli, GARDENA GmbH - Please refer to git log
+ *
+ *******************************************************************************/
+
+/* Examples are taken from (but adjusted):
+ * https://datatracker.ietf.org/doc/html/rfc7959
+ *
+ * "In all these examples, a Block option is shown in a decomposed way
+ * indicating the kind of Block option (1 or 2) followed by a colon, and
+ * then the block number (NUM), more bit (M), and block size exponent
+ * (2**(SZX+4)) separated by slashes."
+ *
+ * BLOCK:NUM/MORE/SIZE
```

Comment:
I think this would improve clarity:

```suggestion
 * BLOCK_OPTION:NUM/MORE/SIZE
```